dbms/src/Storages/S3/FileCache.h

```diff
@@ -201,17 +201,18 @@ class FileCache
 public:
     static void initialize(PathCapacityMetricsPtr capacity_metrics_, const StorageRemoteCacheConfig & config_)
     {
-        global_file_cache_instance = std::make_unique<FileCache>(capacity_metrics_, config_);
+        initializeInstance(capacity_metrics_, config_);
         global_file_cache_initialized.store(true, std::memory_order_release);
     }
 
     static FileCache * instance()
     {
-        return global_file_cache_initialized.load(std::memory_order_acquire) ? global_file_cache_instance.get()
-                                                                             : nullptr;
+        return global_file_cache_initialized.load(std::memory_order_acquire)
+            ? &initializeInstance(nullptr, {}) // Dummy arguments, static variables are already initialized.
+            : nullptr;
     }
 
-    static void shutdown() { global_file_cache_instance = nullptr; }
+    static void shutdown() { instance()->~FileCache(); } // Call destructor to destroy the global instance.
```

Comment:
Why explicitly destruct a global singleton and make it still accessible afterwards? If its lifetime is shorter, make it not accessible after the shutdown.
If some of its resources should be reclaimed before the programs shuts down, then reclaim it in `FileCache` explicitly.

Reply:
Add a new `GlobalFileCacheInstance` to resolve "unique_ptr can't delete an incomplete type"